### ISSUE TITLE
no mop limit crashes ONI with today's patch

### ISSUE THREAD

**Issue by Mr-FLK** [asquared31415/ONI-Mods#40]
Exception while loading mod asquared31415.NoMopLimit at C:/Users/Legitconcern/Documents/Klei/OxygenNotIncluded/mods/Steam/2313125589. 
HarmonyLib.HarmonyException: IL Compile Error ---> System.FormatException: Method virtual System.Void MopTool::OnDragTool(System.Int32 cell, System.Int32 distFromOrigin) cannot be patched. Reason: Invalid IL code in (wrapper dynamic-method) MopTool:MopTool.OnDragTool_Patch0 (MopTool,int,int): IL_0179: ret       

  at HarmonyLib.Memory.DetourMethodAndPersist (System.Reflection.MethodBase original, System.Reflection.MethodBase replacement) [0x00021] in <6dcb326e4f6442999f701f1e67d0b5a0>:0 
  at HarmonyLib.PatchFunctions.UpdateWrapper (System.Reflection.MethodBase original, HarmonyLib.PatchInfo patchInfo) [0x0007b] in <6dcb326e4f6442999f701f1e67d0b5a0>:0 
   --- End of inner exception stack trace ---
  at HarmonyLib.PatchClassProcessor.ReportException (System.Exception exception, System.Reflection.MethodBase original) [0x0010f] in <6dcb326e4f6442999f701f1e67d0b5a0>:0 
  at HarmonyLib.PatchClassProcessor.Patch () [0x00093] in <6dcb326e4f6442999f701f1e67d0b5a0>:0 
  at HarmonyLib.Harmony.<PatchAll>b__10_0 (System.Type type) [0x00007] in <6dcb326e4f6442999f701f1e67d0b5a0>:0 
  at HarmonyLib.CollectionExtensions.Do[T] (System.Collections.Generic.IEnumerable`1[T] sequence, System.Action`1[T] action) [0x00014] in <6dcb326e4f6442999f701f1e67d0b5a0>:0 
  at HarmonyLib.Harmony.PatchAll (System.Reflection.Assembly assembly) [0x00006] in <6dcb326e4f6442999f701f1e67d0b5a0>:0 
  at KMod.UserMod2.OnLoad (HarmonyLib.Harmony harmony) [0x00007] in <043309e6f0914d9f9e207a782760f195>:0 
  at NoMopLimit.NoMopLimitInfo.OnLoad (HarmonyLib.Harmony harmony) [0x0000a] in <151c338d569c4363a46eb8311bb8a73a>:0 
  at KMod.DLLLoader.LoadDLLs (KMod.Mod ownerMod, System.String harmonyId, System.String path, System.Boolean isDev) [0x00222] in <043309e6f0914d9f9e207a782760f195>:0 

  at UnityEngine.Debug.LogError (System.Object message, UnityEngine.Object context) [0x00000] in <72b60a3dd8cd4f12a155b761a1af9144>:0 
  at Debug.LogError (System.Object obj, UnityEngine.Object context) [0x00000] in <3a2e13ca97c146c4b5d2f735a443e5e3>:0 
  at DebugUtil.LogErrorArgs (UnityEngine.Object context, System.Object[] objs) [0x00000] in <3a2e13ca97c146c4b5d2f735a443e5e3>:0 
  at DebugUtil.LogException (UnityEngine.Object context, System.String errorMessage, System.Exception e) [0x00000] in <3a2e13ca97c146c4b5d2f735a443e5e3>:0 
  at KMod.DLLLoader.LoadDLLs (KMod.Mod ownerMod, System.String harmonyId, System.String path, System.Boolean isDev) [0x00000] in <043309e6f0914d9f9e207a782760f195>:0 
  at KMod.Mod.Load (KMod.Content content) [0x00000] in <043309e6f0914d9f9e207a782760f195>:0 
  at KMod.Manager.Load (KMod.Content content) [0x00000] in <043309e6f0914d9f9e207a782760f195>:0 
  at Global.Awake () [0x00000] in <043309e6f0914d9f9e207a782760f195>:0 
  at UnityEngine.Object.Internal_InstantiateSingleWithParent_Injected (UnityEngine.Object data, UnityEngine.Transform parent, UnityEngine.Vector3& pos, UnityEngine.Quaternion& rot) [0x00000] in <72b60a3dd8cd4f12a155b761a1af9144>:0 
  at UnityEngine.Object.Internal_InstantiateSingleWithParent (UnityEngine.Object data, UnityEngine.Transform parent, UnityEngine.Vector3 pos, UnityEngine.Quaternion rot) [0x00000] in <72b60a3dd8cd4f12a155b761a1af9144>:0 
  at UnityEngine.Object.Instantiate (UnityEngine.Object original, UnityEngine.Vector3 position, UnityEngine.Quaternion rotation, UnityEngine.Transform parent) [0x00000] in <72b60a3dd8cd4f12a155b761a1af9144>:0 
  at UnityEngine.Object.Instantiate[T] (T original, UnityEngine.Vector3 position, UnityEngine.Quaternion rotation, UnityEngine.Transform parent) [0x00000] in <72b60a3dd8cd4f12a155b761a1af9144>:0 
  at Util.KInstantiate (UnityEngine.GameObject original, UnityEngine.Vector3 position, UnityEngine.Quaternion rotation, UnityEngine.GameObject parent, System.String name, System.Boolean initialize_id, System.Int32 gameLayer) [0x00000] in <3a2e13ca97c146c4b5d2f735a443e5e3>:0 
  at Util.KInstantiate (UnityEngine.GameObject original, UnityEngine.GameObject parent, System.String name) [0x00000] in <3a2e13ca97c146c4b5d2f735a443e5e3>:0 
  at LaunchInitializer.Update () [0x00000] in <043309e6f0914d9f9e207a782760f195>:0 
Build: U52-622509-SD

**Comment by fruixs** [comment]
I had a same bug

```
Exception while loading mod asquared31415.NoMopLimit at C:/Users/fruix/Documents/Klei/OxygenNotIncluded/mods/Steam/2313125589. 
HarmonyLib.HarmonyException: IL Compile Error ---> System.FormatException: Method virtual System.Void MopTool::OnDragTool(System.Int32 cell, System.Int32 distFromOrigin) cannot be patched. Reason: Invalid IL code in (wrapper dynamic-method) MopTool:MopTool.OnDragTool_Patch0 (MopTool,int,int): IL_0179: ret       


  at HarmonyLib.Memory.DetourMethodAndPersist (System.Reflection.MethodBase original, System.Reflection.MethodBase replacement) [0x00021] in <6dcb326e4f6442999f701f1e67d0b5a0>:0 
  at HarmonyLib.PatchFunctions.UpdateWrapper (System.Reflection.MethodBase original, HarmonyLib.PatchInfo patchInfo) [0x0007b] in <6dcb326e4f6442999f701f1e67d0b5a0>:0 
   --- End of inner exception stack trace ---
  at HarmonyLib.PatchClassProcessor.ReportException (System.Exception exception, System.Reflection.MethodBase original) [0x0010f] in <6dcb326e4f6442999f701f1e67d0b5a0>:0 
  at HarmonyLib.PatchClassProcessor.Patch () [0x00093] in <6dcb326e4f6442999f701f1e67d0b5a0>:0 
  at HarmonyLib.Harmony.<PatchAll>b__10_0 (System.Type type) [0x00007] in <6dcb326e4f6442999f701f1e67d0b5a0>:0 
  at HarmonyLib.CollectionExtensions.Do[T] (System.Collections.Generic.IEnumerable`1[T] sequence, System.Action`1[T] action) [0x00014] in <6dcb326e4f6442999f701f1e67d0b5a0>:0 
  at HarmonyLib.Harmony.PatchAll (System.Reflection.Assembly assembly) [0x00006] in <6dcb326e4f6442999f701f1e67d0b5a0>:0 
  at KMod.UserMod2.OnLoad (HarmonyLib.Harmony harmony) [0x00007] in <043309e6f0914d9f9e207a782760f195>:0 
  at NoMopLimit.NoMopLimitInfo.OnLoad (HarmonyLib.Harmony harmony) [0x0000a] in <151c338d569c4363a46eb8311bb8a73a>:0 
  at KMod.DLLLoader.LoadDLLs (KMod.Mod ownerMod, System.String harmonyId, System.String path, System.Boolean isDev) [0x00222] in <043309e6f0914d9f9e207a782760f195>:0 

  at UnityEngine.Debug.LogError (System.Object message, UnityEngine.Object context) [0x00000] in <72b60a3dd8cd4f12a155b761a1af9144>:0 
  at Debug.LogError (System.Object obj, UnityEngine.Object context) [0x00000] in <3a2e13ca97c146c4b5d2f735a443e5e3>:0 
  at DebugUtil.LogErrorArgs (UnityEngine.Object context, System.Object[] objs) [0x00000] in <3a2e13ca97c146c4b5d2f735a443e5e3>:0 
  at DebugUtil.LogException (UnityEngine.Object context, System.String errorMessage, System.Exception e) [0x00000] in <3a2e13ca97c146c4b5d2f735a443e5e3>:0 
  at KMod.DLLLoader.LoadDLLs (KMod.Mod ownerMod, System.String harmonyId, System.String path, System.Boolean isDev) [0x00000] in <043309e6f0914d9f9e207a782760f195>:0 
  at KMod.Mod.Load (KMod.Content content) [0x00000] in <043309e6f0914d9f9e207a782760f195>:0 
  at KMod.Manager.Load (KMod.Content content) [0x00000] in <043309e6f0914d9f9e207a782760f195>:0 
  at Global.Awake () [0x00000] in <043309e6f0914d9f9e207a782760f195>:0 
  at UnityEngine.Object.Internal_InstantiateSingleWithParent_Injected (UnityEngine.Object data, UnityEngine.Transform parent, UnityEngine.Vector3& pos, UnityEngine.Quaternion& rot) [0x00000] in <72b60a3dd8cd4f12a155b761a1af9144>:0 
  at UnityEngine.Object.Internal_InstantiateSingleWithParent (UnityEngine.Object data, UnityEngine.Transform parent, UnityEngine.Vector3 pos, UnityEngine.Quaternion rot) [0x00000] in <72b60a3dd8cd4f12a155b761a1af9144>:0 
  at UnityEngine.Object.Instantiate (UnityEngine.Object original, UnityEngine.Vector3 position, UnityEngine.Quaternion rotation, UnityEngine.Transform parent) [0x00000] in <72b60a3dd8cd4f12a155b761a1af9144>:0 
  at UnityEngine.Object.Instantiate[T] (T original, UnityEngine.Vector3 position, UnityEngine.Quaternion rotation, UnityEngine.Transform parent) [0x00000] in <72b60a3dd8cd4f12a155b761a1af9144>:0 
  at Util.KInstantiate (UnityEngine.GameObject original, UnityEngine.Vector3 position, UnityEngine.Quaternion rotation, UnityEngine.GameObject parent, System.String name, System.Boolean initialize_id, System.Int32 gameLayer) [0x00000] in <3a2e13ca97c146c4b5d2f735a443e5e3>:0 
  at Util.KInstantiate (UnityEngine.GameObject original, UnityEngine.GameObject parent, System.String name) [0x00000] in <3a2e13ca97c146c4b5d2f735a443e5e3>:0 
  at LaunchInitializer.Update () [0x00000] in <043309e6f0914d9f9e207a782760f195>:0 
Build: U52-622509-SC
```

[Player.log](https://github.com/user-attachments/files/16327704/Player.log)

**Comment by mooklepticon** [comment]
Same.

**Comment by asquared31415** [comment]
This has been resolved in version 2.0.1, just released.  Steam may take several minutes to hours to give you the new update.  If the problem persists, try using [Mod Updater](https://steamcommunity.com/sharedfiles/filedetails/?id=2018291283) to be sure you're getting the update.
